wandb/sdk/wandb_run.py

```diff
@@ -1959,16 +2003,157 @@ def _on_ready(self) -> None:
         # object is about to be returned to the user, don't let them modify it
         self._freeze()
 
+    def _log_job(self) -> None:
+        artifact = None
+        input_types = TypeRegistry.type_of(self.config.as_dict()).to_json()
+        output_types = TypeRegistry.type_of(self.summary._as_dict()).to_json()
+
+        import pkg_resources
+
+        installed_packages_list = sorted(
+            [f"{d.key}=={d.version}" for d in iter(pkg_resources.working_set)]
+        )
+
+        for job_creation_function in [
+            self._create_repo_job,
+            self._create_artifact_job,
+            self._create_image_job,
+        ]:
+            artifact = job_creation_function(
```

Comment:
you probably want to catch the `AssertionError` if you going to try all the options...